IS_INVALID_BUFFER_SIZE when enabling subsampling

Hi all,

I'm using a UI-3591LE-C-HQ with ueye_cam and am having an issue setting the sub_sampling parameter to anything other than 1.  ueyedemo allows subsampling of 2x and 4x, however ueye_cam throws an error for either subsampling option.  Selecting either subsampling option from rqt or the launch file outputs:

[ERROR] [1463407478.498894127]: Could not start free-run live video mode for "[camera]" (IS_INVALID_BUFFER_SIZE)
[ERROR] [1463407478.499014645]: Shutting down driver nodelet for [camera]
[Thread 0x7fffeda5d700 (LWP 5895) exited]
[Thread 0x7fffef260700 (LWP 5888) exited]
[Thread 0x7fffeea5f700 (LWP 5889) exited]
[Thread 0x7fffbaf42700 (LWP 7124) exited]
[Thread 0x7fffdbfff700 (LWP 5909) exited]
[Thread 0x7fffc0ff9700 (LWP 6025) exited]
[Thread 0x7fffc17fa700 (LWP 6022) exited]
[Thread 0x7fffc1ffb700 (LWP 6021) exited]
[Thread 0x7fffc27fc700 (LWP 6005) exited]
[Thread 0x7fffc3fff700 (LWP 5996) exited]
[Thread 0x7fffc37fe700 (LWP 5998) exited]
[Thread 0x7fffc2ffd700 (LWP 5999) exited]
[tcsetpgrp failed in terminal_inferior: Inappropriate ioctl for device]
[tcsetpgrp failed in terminal_inferior: Inappropriate ioctl for device]
[tcsetpgrp failed in terminal_inferior: Inappropriate ioctl for device]
[tcsetpgrp failed in terminal_inferior: Inappropriate ioctl for device]
[Thread 0x7fffd9a98700 (LWP 5992) exited]
[Thread 0x7fffda299700 (LWP 5994) exited]
[Thread 0x7fffd9297700 (LWP 5993) exited]
[Thread 0x7fffe57fa700 (LWP 5904) exited]
[Thread 0x7fffe4ff9700 (LWP 5905) exited]
[Thread 0x7fffed25c700 (LWP 5897) exited]
[Thread 0x7fffe5ffb700 (LWP 5903) exited]
[Thread 0x7fffe6ffd700 (LWP 5901) exited]
[Thread 0x7fffe67fc700 (LWP 5902) exited]
[Thread 0x7fffeca5b700 (LWP 5898) exited]
[Thread 0x7fffe77fe700 (LWP 5900) exited]
[Thread 0x7fffe7fff700 (LWP 5899) exited]
[Thread 0x7fffba741700 (LWP 6391) exited]
[Thread 0x7fffee25e700 (LWP 5890) exited]
[Inferior 1 (process 5870) exited normally]


It looks like a similar (IS_INVALID_BUFFER_SIZE) problem was fixed for binning in https://github.com/anqixu/ueye_cam/commit/2bee82e54e0a303be54e0c949465b968f59a476e.  Any advice on how to fix the issue for subsampling?

--- a/cfg/UEyeCam.cfg
+++ b/cfg/UEyeCam.cfg
@@ -10,9 +10,9 @@ RECONFIGURE_CLOSE = 3
 gen = ParameterGenerator()
 
 gen.add("image_width", int_t, RECONFIGURE_STOP,
-  "Width of camera's area of interest (prior to subsampling, binning, or sensor scaling)", 640, 16, 2560)
+  "Width of camera's area of interest (prior to subsampling, binning, or sensor scaling)", 640, 16, 4912)
 gen.add("image_height", int_t, RECONFIGURE_STOP,
-  "Height of camera's area of interest (prior to subsampling, binning, or sensor scaling)", 480, 4, 1920) 
+  "Height of camera's area of interest (prior to subsampling, binning, or sensor scaling)", 480, 4, 3684) 
 gen.add("image_left", int_t, RECONFIGURE_STOP,
   "Left index for camera's area of interest (-1: center)", -1, -1, 2560-16)
 gen.add("image_top", int_t, RECONFIGURE_STOP,

--- a/src/ueye_cam_driver.cpp
+++ b/src/ueye_cam_driver.cpp
@@ -1074,9 +1074,9 @@ INT UEyeCamDriver::reallocateCamBuffer() {
 
   // Allocate new memory section for IDS driver to use as frame buffer
   INT frameWidth = cam_aoi_.s32Width /
-    (cam_sensor_scaling_rate_ * cam_subsampling_rate_);
+    (cam_sensor_scaling_rate_);
   INT frameHeight = cam_aoi_.s32Height /
-    (cam_sensor_scaling_rate_ * cam_subsampling_rate_);
+    (cam_sensor_scaling_rate_);
   if ((is_err = is_AllocImageMem(cam_handle_, frameWidth, frameHeight,
       bits_per_pixel_, &cam_buffer_, &cam_buffer_id_)) != IS_SUCCESS) {
     ERROR_STREAM("Failed to allocate " << frameWidth << " x " << frameHeight <<

--- a/src/ueye_cam_nodelet.cpp
+++ b/src/ueye_cam_nodelet.cpp
@@ -675,8 +675,8 @@ INT UEyeCamNodelet::syncCamConfig(string dft_mode_str) {
   
   // (Re-)populate ROS image message
   ros_image_.header.frame_id = "/" + frame_name_;
-  ros_image_.height = cam_params_.image_height / (cam_params_.sensor_scaling * cam_params_.subsampling);
-  ros_image_.width = cam_params_.image_width / (cam_params_.sensor_scaling * cam_params_.subsampling);
+  ros_image_.height = cam_params_.image_height / (cam_params_.sensor_scaling);
+  ros_image_.width = cam_params_.image_width / (cam_params_.sensor_scaling);
   ros_image_.encoding = cam_params_.color_mode;
   ros_image_.step = cam_buffer_pitch_;
   ros_image_.is_bigendian = 0;

